common/containers/Tabs/Swap/components/SwapProgress.jsx

```diff
@@ -0,0 +1,141 @@
+//flow
+import React, { Component } from 'react';
+import translate from 'translations';
+import { bityConfig } from 'config/bity';
+
+export type Props = {
+  destinationKind: string,
+  destinationAddress: string,
+  originKind: string,
+  orderStatus: string,
+  // actions
+  showNotification: Function
+};
+
+export default class SwapProgress extends Component {
+  props: Props;
+
+  state: {
+    hasShownViewTx: false
+  };
+
+  computedClass(step: number) {
+    const {
+      orderStatus,
+      destinationKind,
+      showNotification,
+      destinationAddress
+    } = this.props;
+
+    let cssClass = 'progress-item';
+
+    switch (orderStatus) {
+      case 'OPEN':
+        if (step < 2) {
+          return cssClass + ' progress-true';
+        } else if (step === 2) {
+          return cssClass + ' progress-active';
+        } else {
+          return cssClass;
+        }
+      case 'RCVE':
+        if (step < 4) {
+          return cssClass + ' progress-true';
+        } else if (step === 4) {
+          return cssClass + ' progress-active';
+        } else {
+          return cssClass;
+        }
+      case 'FILL':
+        if (step < 5) {
+          cssClass += ' progress-true';
+        } else if (step === 5) {
+          cssClass += ' progress-active';
+        }
+
+        if (!this.state.hasShownViewTx) {
+          let linkElement;
+          let link;
+          // everything but BTC is a token
+          if (destinationKind !== 'BTC') {
+            link = bityConfig.ethExplorer.replace(
+              '[[txHash]]',
+              destinationAddress
+            );
+            linkElement = `<a href="${link}" target='_blank' rel='noopener'> View your transaction </a>`;
+            // BTC uses a different explorer
+          } else {
+            link = link = bityConfig.btcExplorer.replace(
+              '[[txHash]]',
+              destinationAddress
+            );
+            linkElement = `<a href="${link}" target='_blank' rel='noopener'> View your transaction </a>`;
+          }
+
+          this.setState({ hasShownViewTx: true }, () => {
+            showNotification('success', linkElement);
```

Comment:
I'm not sure how showNotification works, but is it possible to use JSX instead of strings? Would be nice to nix any uses of setInnerHTMLDangerously since that's ideally only used for server-side HTML.